interface.go

```diff
@@ -5,20 +5,42 @@ import (
 	"time"
 
 	inet "github.com/libp2p/go-libp2p-net"
-	peer "github.com/libp2p/go-libp2p-peer"
+	"github.com/libp2p/go-libp2p-peer"
 )
 
+// ConnManager tracks connections to peers, and allows consumers to associate metadata
+// with each peer.
+//
+// It enables connections to be trimmed based on implementation-defined heuristics.
 type ConnManager interface {
+
+	// TagPeer sets a tag value for the given peer.
+	// A tag can be a distance, latency, reputation, etc. value.
```

Comment:
Tag peer really:

1. Tags the peer with the string.
2. Associates a weight with that tag.

latency, distance, etc are probably not good weights.

Reply:
👍 Applied the changes.